packages/strapi-hook-bookshelf/lib/utils/connectivity.js

```diff
@@ -1,94 +1,71 @@
 'use strict';
 
-// Node.js core.
-const path = require('path');
-
 // Public node modules
 const inquirer = require('inquirer');
-const rimraf = require('rimraf');
-
-module.exports = (scope, success, error) => {
-  if (scope.client.database === 'sqlite') {
-    return success();
-  }
 
-  let knex;
+const selectQueries = {
+  postgres: "SELECT tablename FROM pg_tables WHERE schemaname='public'",
+  mysql: 'SELECT * FROM information_schema.tables',
+  sqlite: 'select * from sqlite_master',
+};
 
-  try {
-    // eslint-disable-next-line import/no-unresolved
-    knex = require('knex');
-  } catch (err) {
-    // eslint-disable-next-line import/no-unresolved
-    knex = require(path.resolve(scope.tmpPath, 'node_modules', 'knex'));
-  }
+module.exports = async ({ scope, connection }) => {
+  const knex = require('knex');
 
-  // eslint-disable-next-line import/no-unresolved
+  const { settings } = connection;
   const client = knex({
-    client: scope.client.module,
-    connection: Object.assign({}, scope.database.settings, {
-      user: scope.database.settings.username
+    client: settings.client,
+    connection: Object.assign({}, settings, {
+      user: settings.username,
     }),
-    useNullAsDefault: true
+    useNullAsDefault: true,
   });
 
-  client.raw('select 1+1 as result').then(() => {
-    const selectQueries = {
-      postgres: 'SELECT tablename FROM pg_tables WHERE schemaname=\'public\'',
-      mysql: 'SELECT * FROM information_schema.tables',
-      sqlite: 'select * from sqlite_master'
-    };
+  const destroyClientAndThrow = err => {
+    return client.destroy().then(
+      () => {
+        throw err;
+      },
+      () => {
+        throw err;
+      }
+    );
+  };
 
-    client.raw(selectQueries[scope.client.database]).then((tables) => {
-      client.destroy();
+  await client.raw('select 1+1 as result').catch(destroyClientAndThrow);
 
-      const next = () => {
-        rimraf(scope.tmpPath, (err) => {
-          if (err) {
-            console.log(`Error removing connection test folder: ${scope.tmpPath}`);
-          }
+  return client
+    .raw(selectQueries[settings.client])
+    .then(tables => {
+      if (tables.rows && tables.rows.length === 0) {
+        return;
+      }
 
-          success();
-        });
-      };
+      if (scope.dbforce) {
```

Comment:
Interesting that even though I had a clean database on MySQL I still got this:
![image](https://user-images.githubusercontent.com/8593673/60792838-47110f00-a11b-11e9-9594-7c54e33124ad.png)